crates/nu-cli/tests/completions/mod.rs

```diff
@@ -363,6 +400,146 @@ fn file_completions() {
+        Signature::build("list").rest("args", SyntaxShape::Any, "rest of the arguments")
+    }
+
+    fn run(
+        &self,
+        _engine_state: &EngineState,
+        _stack: &mut Stack,
+        _call: &Call,
+        _input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        todo!()
+    }
+}
+
+#[test]
+fn custom_command_completions() {
+    // Create a new engine
+    let (dir, dir_str, mut engine, stack) = new_engine();
+    let delta = {
+        let mut working_set = StateWorkingSet::new(&engine);
+        working_set.add_decl(Box::new(CustomCommandWithAnyArgs));
+        working_set.render()
+    };
+    engine.merge_delta(delta).expect("Failed to merge delta");
+
+    // Instantiate a new completer
+    let mut completer = NuCompleter::new(Arc::new(engine), Arc::new(stack));
+
+    // Test completions for the current folder
+    let target_dir = format!("list {dir_str}{MAIN_SEPARATOR}");
+    let suggestions = completer.complete(&target_dir, target_dir.len());
+
+    // Create the expected values
+    let expected_paths: Vec<String> = vec![
+        folder(dir.join("another")),
+        file(dir.join("custom_completion.nu")),
+        folder(dir.join("directory_completion")),
+        file(dir.join("nushell")),
+        folder(dir.join("test_a")),
+        folder(dir.join("test_b")),
+        file(dir.join(".hidden_file")),
+        folder(dir.join(".hidden_folder")),
+    ];
+
+    #[cfg(windows)]
+    {
+        let separator = '/';
+        let target_dir = format!("list {dir_str}{separator}");
+        let slash_suggestions = completer.complete(&target_dir, target_dir.len());
+
+        let expected_slash_paths: Vec<String> = expected_paths
+            .iter()
+            .map(|s| s.replace('\\', "/"))
+            .collect();
+
+        match_suggestions(&expected_slash_paths, &slash_suggestions);
+    }
+
+    // Match the results
+    match_suggestions(&expected_paths, &suggestions);
+
+    let target_dir = format!("list somefile.txt {dir_str}{MAIN_SEPARATOR}");
+    let suggestions = completer.complete(&target_dir, target_dir.len());
+
+    // Create the expected values
+    let expected_paths: Vec<String> = vec![
+        folder(dir.join("another")),
+        file(dir.join("custom_completion.nu")),
+        folder(dir.join("directory_completion")),
+        file(dir.join("nushell")),
+        folder(dir.join("test_a")),
+        folder(dir.join("test_b")),
+        file(dir.join(".hidden_file")),
+        folder(dir.join(".hidden_folder")),
+    ];
+
+    #[cfg(windows)]
+    {
+        let separator = '/';
+        let target_dir = format!("list {dir_str}{separator}");
+        let slash_suggestions = completer.complete(&target_dir, target_dir.len());
+
+        let expected_slash_paths: Vec<String> = expected_paths
+            .iter()
+            .map(|s| s.replace('\\', "/"))
+            .collect();
+
+        match_suggestions(&expected_slash_paths, &slash_suggestions);
+    }
```

Comment:
Is this necessary? This test is about making sure that there are completions for custom commands. File completion is being tested in the other test above.

Reply:
I will align with the testing culture of this project. So please point this out if I am not. 

When that is said, then my idea here is to protect against regression. Is there another test that checks that the first argument of a custom command also get file completed?

Reply:
You're right, I couldn't find any tests confirming that the first argument of a custom command gets file completed. I'm not sure that's something we need to test, but it doesn't hurt to have, I guess.

But I made this comment specifically about the `#[cfg(windows)] {...}` block, not the whole test. That bit is to make sure that `/` is used as the separator on Windows if the user types in `foo/`. It seems unlikely to me that this behavior would break for custom commands but not for internal commands.

Reply:
Ahh good point. I deleted them.